### DISCUSSION TITLE
Battery powered dead node status

### DISCUSSION THREAD

**Discussion by Gis70** [zwave-js/zwave-js#3678] [Q&A]
Hi, I wonder if the way to determine battery powered nodes status is working as expected.

For example, I made a script to ping node every day and then check theirs status.

I had battery powered devices which didn't responde since 5 days (wake up setted every 24 hours) and the node status after a ping is : node asleep.

What is the algorythm ?

Can't we base the algorythm calculating if wake up delay is overtaken for battery powered devices and then mark them dead ?

Thank you for you help to make my system more reliable.

**Comment by AlCalzone** [comment]
> I made a script to ping node every day and then check theirs status.

That only works for listening nodes, not sleeping ones. Those only wake up for a very short time, send their wakeup messages and then sleep for the rest of the wakeup cycle.

> Can't we base the algorythm calculating if wake up delay is overtaken for battery powered devices and then mark them dead ?

Right now the logic in the driver considers asleep and dead to be states without a transition between them, meaning a sleeping device will never be considered dead.
You could use that approach to detect problematic nodes though. But consider that there might be delays in communication, one-time connectivity issues, etc. If you're too strict, you might end up with false positives.
